src/many-ledger/src/module.rs

```diff
@@ -236,10 +235,22 @@ impl ledger::LedgerCommandsModuleBackend for LedgerModuleImpl {
         } = args;
 
         let from = from.as_ref().unwrap_or(sender);
-
-        // TODO: allow some ACLs or delegation on the ledger.
         if from != sender {
-            return Err(error::unauthorized());
+            if let Some(account) = self.storage.get_account(from) {
+                if account
+                    .features
+                    .has_id(account::features::ledger::AccountLedger::ID)
+                {
+                    account.needs_role(
+                        sender,
+                        [account::Role::Owner, account::Role::CanLedgerTransact],
```

Comment:
nit: We should document somewhere that the account only requires ONE of the listed roles to be granted access.

Reply:
Yeah we should refactor the error to be more explicit.